4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-systems-input-sdl3-gamepad-input-driver.md

---
type: entry
section: systems
subsection: input
rank: 0
importance: 2
anchor: sdl3-gamepad-input-driver
title: SDL3 gamepad input driver
blockquote: Gamepads galore!
text: |

>   Gamepads are nowadays a given in modern PC gaming. Users expect their plugged one to just work. Not only that, but in order to deliver unique experiences, some gamepads are introducing new features.

Reviewer: ```suggestion
  Gamepads are nowadays a given in modern PC gaming. Users expect their gamepad to just plug in and work. Not only that, but in order to deliver unique experiences, some gamepads are introducing new features.
```

>   Gamepads are nowadays a given in modern PC gaming. Users expect their gamepad to just plug in and work. Not only that, but in order to deliver unique experiences, some gamepads are introducing new features.

Reviewer 2: ```suggestion
  Gamepads are a given in modern PC gaming. Users expect their gamepad to just plug in and work. Not only that, but in order to deliver unique experiences, some gamepads are introducing new features.
```
Could a few of these new features for unique experiences be listed? (Adaptive triggers, touchpads, etc?) This sentence piques my interest about them, but then doesn't follow up with details.

On the other hand, given that 4.5 doesn't actually directly support those new features, I could see it being misleading to bring them up

Author: Replaced with 
>   Gamepads are a given in modern PC gaming. Users expect their gamepad to just plug in and work. Not only that, but in order to deliver unique experiences, some gamepads are introducing new features; from adaptive triggers to advanced haptic feedback, as well as microphones and motion controls.

>   Gamepads are a given in modern PC gaming. Users expect their gamepad to just plug in and work. Not only that, but in order to deliver unique experiences, some gamepads are introducing new features; from adaptive triggers to advanced haptic feedback, as well as microphones and motion controls.

Reviewer: ```suggestion
  Gamepads are a given in modern PC gaming. Users expect their gamepad to just plug in and work. Not only that, but in order to deliver unique experiences, some gamepads are introducing new features; from adaptive triggers and advanced haptic feedback, to microphones and motion controls.
```
Flows nicely.

>   Over time, issues accumulated in our gamepad driver implementation and missing features began to creep up. We were facing an ever-growing mountain.

Reviewer 3: ```suggestion
  Over time, issues accumulated in our gamepad driver implementation and missing features began to crop up. We were facing an ever-growing mountain.
```
I'm not sure features would creep up, rather they'd start to get more numerous

>   That's we turned our heads to the SDL project. SDL is a well established and mature cross-platform library that handles audio, keyboard, mouse, joystick, and graphics. We determined that it's now a net positive to defer the responsibility for gamepad handling to it.

Reviewer 3: ```suggestion
  That’s why we turned our heads to the SDL project. SDL is a well established and mature cross-platform library that handles audio, keyboard, mouse, joystick, and graphics. We determined that it's now a net positive to defer the responsibility for gamepad handling to it.
```

>   That's we turned our heads to the SDL project. SDL is a well established and mature cross-platform library that handles audio, keyboard, mouse, joystick, and graphics. We determined that it's now a net positive to defer the responsibility for gamepad handling to it.

Reviewer: ```suggestion
  That's when we turned our heads to the SDL project. SDL is a well established and mature cross-platform library that handles audio, keyboard, mouse, joystick, and graphics. We determined that it's now a net positive to defer the responsibility for gamepad handling to it.
```

Author: Used "why" instead, [as suggested](https://github.com/godotengine/godot-website/pull/1138#discussion_r2267450359) by @AThousandShips

>   That’s why we turned our heads to the SDL project. SDL is a well established and mature cross-platform library that handles audio, keyboard, mouse, joystick, and graphics. We determined that it’s now a net positive to defer the responsibility for gamepad handling to it.

Reviewer 2: ```suggestion
  That’s why we turned our heads to the [SDL project](https://wiki.libsdl.org/SDL3/FrontPage). SDL is a well established and mature cross-platform library that handles audio, keyboard, mouse, joystick, and graphics. We determined that it’s now a net positive to defer the responsibility for gamepad handling to it.
```
Add a link to SDL, since we're discussing it

Reviewer: I agree we should link, but I think linking to the simpler https://libsdl.org/ URL would be better.

>   While this change doesn't by itself bring new features, expect bugfixes and new features to come a little bit faster from now on.

Reviewer 3: ```suggestion
  While this change doesn’t by itself bring new features, expect bugfixes and new features to come a little bit faster from now on.
```
